coderd/database/queries/prebuilds.sql

```diff
@@ -0,0 +1,120 @@
+FROM templates t
+		 INNER JOIN template_versions tv ON tv.template_id = t.id
+		 INNER JOIN template_version_presets tvp ON tvp.template_version_id = tv.id
+		 INNER JOIN template_version_preset_prebuilds tvpp ON tvpp.preset_id = tvp.id
+WHERE (t.id = sqlc.narg('template_id')::uuid OR sqlc.narg('template_id') IS NULL);
+
+-- name: GetRunningPrebuilds :many
+SELECT p.id               AS workspace_id,
+	   p.name             AS workspace_name,
+	   p.template_id,
+	   b.template_version_id,
+	   tvp_curr.id        AS current_preset_id,
+	   CASE
+		   WHEN p.lifecycle_state = 'ready'::workspace_agent_lifecycle_state THEN TRUE
+		   ELSE FALSE END AS ready,
+	   p.created_at
+FROM workspace_prebuilds p
+		 INNER JOIN workspace_latest_build b ON b.workspace_id = p.id
+		 INNER JOIN provisioner_jobs pj ON b.job_id = pj.id
+		 INNER JOIN templates t ON p.template_id = t.id
+		 LEFT JOIN template_version_presets tvp_curr
+				   ON tvp_curr.id = p.current_preset_id -- See https://github.com/coder/internal/issues/398.
+WHERE (b.transition = 'start'::workspace_transition
+	AND pj.job_status = 'succeeded'::provisioner_job_status);
+
+-- name: GetPrebuildsInProgress :many
+SELECT t.id AS template_id, wpb.template_version_id, wpb.transition, COUNT(wpb.transition)::int AS count
+FROM workspace_latest_build wlb
+		 INNER JOIN provisioner_jobs pj ON wlb.job_id = pj.id
+		 INNER JOIN workspace_prebuild_builds wpb ON wpb.id = wlb.id
+		 INNER JOIN templates t ON t.active_version_id = wlb.template_version_id
+WHERE pj.job_status IN ('pending'::provisioner_job_status, 'running'::provisioner_job_status)
+GROUP BY t.id, wpb.template_version_id, wpb.transition;
+
+-- name: GetPresetsBackoff :many
+WITH filtered_builds AS (
+	-- Only select builds which are for prebuild creations
+	SELECT wlb.*, tvp.id AS preset_id, pj.job_status, tvpp.desired_instances
+	FROM template_version_presets tvp
+			 JOIN workspace_latest_build wlb ON wlb.template_version_preset_id = tvp.id
+			 JOIN provisioner_jobs pj ON wlb.job_id = pj.id
+			 JOIN template_versions tv ON wlb.template_version_id = tv.id
+			 JOIN templates t ON tv.template_id = t.id AND t.active_version_id = tv.id
+			 JOIN template_version_preset_prebuilds tvpp ON tvpp.preset_id = tvp.id
+	WHERE wlb.transition = 'start'::workspace_transition),
+	 latest_builds AS (
+		 -- Select only the latest build per template_version AND preset
+		 SELECT fb.*,
+				ROW_NUMBER() OVER (PARTITION BY fb.template_version_preset_id ORDER BY fb.created_at DESC) as rn
+		 FROM filtered_builds fb),
+	 failed_count AS (
+		 -- Count failed builds per template version/preset in the given period
+		 SELECT preset_id, COUNT(*) AS num_failed
+		 FROM filtered_builds
+		 WHERE job_status = 'failed'::provisioner_job_status
+		   AND created_at >= @lookback::timestamptz
+		 GROUP BY preset_id)
+SELECT lb.template_version_id,
+	   lb.preset_id,
+	   MAX(lb.job_status)::provisioner_job_status AS latest_build_status,
```

Comment:
Do we need `MAX(lb.job_status)` here? Can we use `lb.job_status` instead? We already have `GROUP BY lb.job_status`.

Reply:
We can use `lb.job_status`, but I decided to remove it completely, because it always return `job_status=failed`